lib/src/wiggle_abi/req_impl.rs

```diff
@@ -598,16 +598,19 @@ impl FastlyHttpReq for Session {
     ) -> Result<(u32, ResponseHandle, BodyHandle), Error> {
         let handle: PendingRequestHandle = pending_req_handle.into();
 
-        let outcome = match self.async_item_mut(handle.into())?.is_some() {
-            true => {
-                let resp = self
-                    .take_pending_request(pending_req_handle)?
-                    .recv()
-                    .await?;
-                let (resp_handle, resp_body_handle) = self.insert_response(resp);
-                (1, resp_handle, resp_body_handle)
-            }
-            false => (0, INVALID_REQUEST_HANDLE.into(), INVALID_BODY_HANDLE.into()),
+        let outcome = match self.async_item_mut(handle.into())? {
+            Some(item) => match item.is_ready() {
+                true => {
+                    let resp = self
+                        .take_pending_request(pending_req_handle)?
+                        .recv()
+                        .await?;
+                    let (resp_handle, resp_body_handle) = self.insert_response(resp);
+                    (1, resp_handle, resp_body_handle)
+                }
+                false => (0, INVALID_REQUEST_HANDLE.into(), INVALID_BODY_HANDLE.into()),
+            },
+            None => (0, INVALID_REQUEST_HANDLE.into(), INVALID_BODY_HANDLE.into()),
```

Comment:
Could you also please make `async_item_mut` not return an `Option`, so we don't have to deal with this redundant case?